internal/testutil/engine.go

```diff
@@ -0,0 +1,26 @@
+package testutil
+
+import (
+	"fmt"
+	"sync"
+)
+
+var (
+	nestedEngineCount   uint8 = 1
+	nestedEngineCountMu sync.Mutex
+)
+
+// returns a device name and cidr to use; enables us to have unique devices+ip ranges for nested
+// engine services to prevent conflicts
+func GetUniqueNestedEngineNetwork() (deviceName string, cidr string) {
```

Comment:
:tada:

255 tests ought to be enough for everyone :slightly_smiling_face:

Reply:
Haha yeah I considered doing some bit fiddling to make it a uint16 and use the other part of the IP range, but figured this should last us for a good while and will be happy to know once we've run 255+ tests with nested engines 😄